lib/connectionConfig/connectionConfig.ts

```diff
@@ -112,25 +113,25 @@ export namespace ConnectionConfig {
    */
   export function validate(config: ConnectionConfig, options?: ConnectionConfigOptions): void {
     if (!options) options = {};
-    if (!config || (config && typeof config !== "object")) {
-      throw new Error("'config' is a required parameter and must be of type: 'object'.");
-    }
-    if (!config.endpoint || (config.endpoint && typeof config.endpoint !== "string")) {
-      throw new Error("'endpoint' is a required property of ConnectionConfig.");
-    }
-    if (config.entityPath && typeof config.entityPath !== "string") {
-      throw new Error("'entityPath' must be of type 'string'.");
-    }
-    if (options.isEntityPathRequired && !config.entityPath) {
-      throw new Error("'entityPath' is a required property of ConnectionConfig.");
-    }
-    if (!config.sharedAccessKeyName ||
-      (config.sharedAccessKeyName && typeof config.sharedAccessKeyName !== "string")) {
-      throw new Error("'sharedAccessKeyName' is a required property of ConnectionConfig.");
+
```

Comment:
In Event Hubs and Service Bus, we are calling `validate` from the static methods that create EventHubsClient/ServiceBusClient.

Instead, we should call it from the constructor so that we also cover the case where user calls the constructor directly. Can you please log an issue for that?

Reply:
I'll create a issue for this.

Just FYI, in EventHubs we are not calling `validate` anywhere.